main.go

```diff
@@ -186,13 +184,14 @@ func main() {
 		InvoicePubSub:  service.NewPubsub(),
 	}
 
+	logMw := createLoggingMiddleware(logger)
 	// strict rate limit for requests for sending payments
 	strictRateLimitMiddleware := createRateLimitMiddleware(c.StrictRateLimit, c.BurstRateLimit)
-	secured := e.Group("", tokens.Middleware(c.JWTSecret))
-	securedWithStrictRateLimit := e.Group("", tokens.Middleware(c.JWTSecret), strictRateLimitMiddleware)
+	secured := e.Group("", tokens.Middleware(c.JWTSecret), logMw)
+	securedWithStrictRateLimit := e.Group("", tokens.Middleware(c.JWTSecret), strictRateLimitMiddleware, logMw)
 
-	RegisterLegacyEndpoints(svc, e, secured, securedWithStrictRateLimit, strictRateLimitMiddleware, tokens.AdminTokenMiddleware(c.AdminToken))
-	RegisterV2Endpoints(svc, e, secured, securedWithStrictRateLimit, strictRateLimitMiddleware, tokens.AdminTokenMiddleware(c.AdminToken))
+	RegisterLegacyEndpoints(svc, e, secured, securedWithStrictRateLimit, strictRateLimitMiddleware, tokens.AdminTokenMiddleware(c.AdminToken), logMw)
```

Comment:
do we need to pass the middleware to the individual endpoints? isn't it enough to have it on the group? (after the JWT middleware)

Reply:
This is needed because these functions also register unauthenticated endpoints (/auth and /create). Otherwise it would have been enough to have it on the group.

Reply:
There is some code duplication here but that can be cleaned up later I guess.

Reply:
but if the middleware works for authenticated and unauthenticated endpoints why do we need to pass it to each individually?

Reply:
ok, this is that those endpoints also log at all, otherwise there would not be any logs there I guess.